learningobjects/api.py

```diff
@@ -0,0 +1,39 @@
+"""
+Helper functions for using the models, so external
+apps don't tie functionality to internal implementation.
+"""
+
+from datetime import datetime
+
+from learningobjects.models import Course, Repository
+
+# pylint: disable=no-member
+
+def create_course(org, course_number, semester, user_id):
+    """
+    Add a course to the database.
+    """
+    # TODO: Determine rules for accepting/requiring a repository.
+    kwargs = {
+        "org": org, "course_number": course_number, "semester": semester,
+        "import_date": datetime.now(), 'imported_by_id': user_id,
+        "repository_id": get_temp_repository(user_id).id
+    }
+    courses = Course.objects.filter(**kwargs)
+    if courses.count() > 0:
+        return None, "course already exists"
+    return Course.objects.create(**kwargs), ""
```

Comment:
Yeah, my stance is evidently misunderstood. I never advocated never handling models. I simply meant that functions like `create_course` should exist and be used rather than having the same logic simply in a view or whatever. Because it could change, and it would certainly end up being non-DRY.